src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/ElytraCapListener.java

```diff
@@ -0,0 +1,54 @@
+package io.github.thebusybiscuit.slimefun4.implementation.listeners;
+
+import io.github.thebusybiscuit.slimefun4.core.attributes.DamageableItem;
+import io.github.thebusybiscuit.slimefun4.core.attributes.ProtectionType;
+import io.github.thebusybiscuit.slimefun4.core.attributes.ProtectiveArmor;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import io.github.thebusybiscuit.slimefun4.implementation.items.armor.ElytraCap;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+import me.mrCookieSlime.Slimefun.api.Slimefun;
+import org.bukkit.GameMode;
+import org.bukkit.Sound;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityDamageEvent;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.Nonnull;
+import java.util.Arrays;
+
+/**
+ * The {@link Listener} for the {@link ElytraCap}.
+ *
+ * @author Seggan
+ */
+public class ElytraCapListener implements Listener {
+
+    public ElytraCapListener(@Nonnull SlimefunPlugin plugin) {
+        plugin.getServer().getPluginManager().registerEvents(this, plugin);
+    }
+
+    @EventHandler
+    public void onPlayerCrash(EntityDamageEvent e) {
+        if (!(e.getEntity() instanceof Player)) return;
+        if (!(e.getCause() == EntityDamageEvent.DamageCause.FALL ||
+            e.getCause() == EntityDamageEvent.DamageCause.FLY_INTO_WALL)) return;
+
+        Player p = (Player) e.getEntity();
+        if (p.isGliding()) {
+            ItemStack stack = p.getInventory().getHelmet();
+            SlimefunItem item = SlimefunItem.getByItem(stack);
+            if (!Slimefun.hasUnlocked(p, item, true)) return;
+            if (item instanceof ProtectiveArmor) {
+                if (!Arrays.asList(((ProtectiveArmor) item).getProtectionTypes())
+                    .contains(ProtectionType.FLYING_INTO_WALL)) return;
```

Comment:
No no no no, there is a method like `ProtectiveArmor#hasProtectionAgainst()` or so this should be used.
And please stop with this excessive use of `return;` statements and inverse conditions, that's not clean at all.

Reply:
So invert the condition and finger to normal if?

Reply:
i do not see any ProtectiveArmor#hasProtectionAgainst()

Reply:
It may be in Slimefun class or sth. Check hazmat suit code.

Reply:
@Seggan PlayerProfile#hasFullProtectionAgainst()
https://github.com/Slimefun/Slimefun4/blob/master/src/main/java/io/github/thebusybiscuit/slimefun4/api/player/PlayerProfile.java#L471